### ISSUE TITLE
Question about the way of optimization behind Xgboost

### ISSUE THREAD

**Issue by Orbean** [dmlc/xgboost#3213]
In Xgboost, hessian( in other words, second-order Taylor expansion of the loss function) is used 
to optimize the loss function, which is Newton's method. I have several questions about this point.

1) As long as I know, in the case that hessian matrix is not positive value(ttps://en.wikipedia.org/wiki/Positive-definite_matrix), this method, Newton's method, does not work bcause it would lead local maximum. How does Xgboost avoid this problem? 
2) Generally speaking, computing hessian matrix is very memory-consuming. But XGboost is so fast( faster than GBRT from scikit-learn?) What made it so fast?

Thank you.

**Comment by khotilov** [comment]
During split-finding while building the trees it's univariate greedy optimization.

**Comment by Orbean** [comment]
oh so it's different from newton's method?
Why does it need the 2nd order Taylor expansion?

**Comment by hcho3** [comment]
With hessian statistics, the constructed tree can better approximate the residuals between true and predicted labels.
